lib/config/config-initializer.js

```diff
@@ -45,44 +50,171 @@ function writeFile(config, format, callback) {
         return;
     }
 
-    // install any external configs as well as any included plugins
-    if (config.extends && config.extends.indexOf("eslint") === -1) {
-        console.log("Installing additional dependencies");
-        exec("npm i eslint-config-" + config.extends + " --save-dev", function(err) {
+    callback();
+}
 
-            if (err) {
-                callback(err);
-                return;
-            }
+/**
+ * Synchronously install necessary plugins, configs, parsers, etc. based on the config
+ * @param   {Object} config  config object
+ * @returns {void}
+ */
+function installModules(config) {
+    var modules = [],
+        installStatus,
+        modulesToInstall;
 
-            // TODO: consider supporting more than 1 plugin though it's required yet.
-            exec("npm i eslint-plugin-" + config.plugins[0] + " --save-dev", callback);
-        });
-        return;
+    // Create a list of modules which should be installed based on config
+    if (config.plugins) {
+        modules = modules.concat(config.plugins.map(function(name) {
+            return "eslint-plugin-" + name;
+        }));
+    }
+    if (config.extends && config.extends.indexOf("eslint:") === -1) {
+        modules.push(config.extends);
+    }
+    if (config.parser && config.parser !== "espree") {
+        modules.push(config.parser);
     }
 
-    // install the react plugin if it was explictly chosen
-    if (config.plugins && config.plugins.indexOf("react") >= 0) {
-        console.log("Installing React plugin");
-        exec("npm i eslint-plugin-react --save-dev", callback);
+    // Determine which modules are already installed
+    if (modules.length === 0) {
         return;
     }
-    callback();
+    installStatus = npmUtil.checkInstalled(modules);
+
+    // Install packages which aren't already installed
+    modulesToInstall = Object.keys(installStatus)
+    .filter(function(module) {
+        return installStatus[module] === false;
+    });
```

Comment:
Okay.  I've been trying to write more "functional" code, and chaining on separate lines is common in that style, and it helps to avoid long lines.  But, I don't mind changing it either.